datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -418,4 +420,11 @@ private String getAccountIdFromCrn(String userCrn) {
     private DetailedEnvironmentResponse getEnvironment(SdxClusterRequest sdxClusterRequest) {
         return environmentClientService.getByName(sdxClusterRequest.getEnvironment());
     }
+
+    private String trimBaseLocation(String baseLocation) {
+        if (baseLocation.endsWith("/")) {
```

Comment:
method content could be changed to Stringutils.substringBeforeLast().

Reply:
Again, bullshit! :D

Reply:
The following base locations will produces wrong results: 
* s3a://some/dir ->s3a://some
* s3a://some -> s3a:/

This function only trims the last / if exist otherwise return the unmodified string.